### PR TITLE
Update helm to v3.9.0 and k8s to v1.24

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Updates helm to v3.9.0 and updates all k8s dependencies to v1.24.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

xref #195 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Currently only `make build` -- additional testing in progress.

=== UPDATE ===

Verified that with 472 CRDs installed, that I get client-side rate limiting with `up` build prior to this PR, and with this PR `up uxp upgrade` takes ~3.5 seconds.

```
🤖 (up) up --version
v0.12.0-rc.0.39.g7d1e248
🤖 (up) k get crds | wc -l
472
🤖 (up) k get pkg
NAME                                                INSTALLED   HEALTHY   PACKAGE                                         AGE
provider.pkg.crossplane.io/upbound-provider-aws     True        True      xpkg.upbound.io/upbound/provider-aws:v0.4.0     2m44s
provider.pkg.crossplane.io/upbound-provider-azure   True        True      xpkg.upbound.io/upbound/provider-azure:v0.4.0   2m52s
provider.pkg.crossplane.io/upbound-provider-gcp     True        True      xpkg.upbound.io/upbound/provider-gcp:v0.4.0     2m39s
🤖 (up) k get pkg
NAME                                                INSTALLED   HEALTHY   PACKAGE                                         AGE
provider.pkg.crossplane.io/upbound-provider-aws     True        True      xpkg.upbound.io/upbound/provider-aws:v0.4.0     2m58s
provider.pkg.crossplane.io/upbound-provider-azure   True        True      xpkg.upbound.io/upbound/provider-azure:v0.4.0   3m6s
provider.pkg.crossplane.io/upbound-provider-gcp     True        True      xpkg.upbound.io/upbound/provider-gcp:v0.4.0     2m53s
🤖 (up) k get pods -A
NAMESPACE            NAME                                                   READY   STATUS    RESTARTS   AGE
kube-system          coredns-558bd4d5db-l6xt9                               1/1     Running   0          4m27s
kube-system          coredns-558bd4d5db-mbgq9                               1/1     Running   0          4m27s
kube-system          etcd-kind-control-plane                                1/1     Running   0          4m31s
kube-system          kindnet-gdfrs                                          1/1     Running   0          4m27s
kube-system          kube-apiserver-kind-control-plane                      1/1     Running   0          4m45s
kube-system          kube-controller-manager-kind-control-plane             1/1     Running   0          4m31s
kube-system          kube-proxy-kjpm7                                       1/1     Running   0          4m27s
kube-system          kube-scheduler-kind-control-plane                      1/1     Running   0          4m31s
local-path-storage   local-path-provisioner-547f784dff-228wg                1/1     Running   0          4m27s
upbound-system       crossplane-75dc9ddfc8-xx9jn                            1/1     Running   0          4m26s
upbound-system       crossplane-rbac-manager-68b5bd6d6d-j6ns6               1/1     Running   0          4m26s
upbound-system       upbound-bootstrapper-67cb7446cf-tnmt7                  1/1     Running   0          4m26s
upbound-system       upbound-provider-aws-5f64f1c0acc0-6cc56569db-h5nn7     1/1     Running   0          75s
upbound-system       upbound-provider-azure-38e9e72b6c6e-85dd946d47-dj6tr   1/1     Running   0          101s
upbound-system       upbound-provider-gcp-a5f5934add58-7b9b5bb6c6-9w7d6     1/1     Running   0          108s
upbound-system       xgql-7585dd746b-zmqtt                                  1/1     Running   1          4m26s
🤖 (up) up uxp upgrade
I0711 16:28:02.865333   90176 request.go:665] Waited for 1.194986223s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:44995/apis/sqs.aws.upbound.io/v1beta1?timeout=32s
I0711 16:28:12.910916   90176 request.go:665] Waited for 3.597625925s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:44995/apis/s3.aws.upbound.io/v1beta1?timeout=32s
🤖 (up) up --version
v0.12.0-rc.0.43.gcfb399d
🤖 (up) time up uxp upgrade

real	0m3.623s
user	0m5.816s
sys	0m0.560s
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -224,4 +227,19 @@ replace (
 	github.com/docker/docker => github.com/moby/moby v17.12.0-ce-rc1.0.20200618181300-9dc6525e6118+incompatible
 	github.com/golang/tools => ./internal/vendor/golang.org/x/tools
 	github.com/googleapis/gnostic => github.com/google/gnostic v0.5.5
+
+	// The following replacements should be kept in sync with upstream
+	// Kubernetes version.
```
Comment (hasheddan) [inline go.mod:232]: These match https://github.com/kubernetes/kubernetes/blob/4ce5a8954017644c5420bae81d72b09b735c21f0/go.mod#L394

Comment (tnthornton) [inline go.mod:232]: I assume this means any time we update our k8s.io dependencies we also need to align this list of replacements. Any concerns with the versions of k8s up is to support with taking this approach? Or do we plan to just deprecate/remove API support in tandem with k8s?

**Review by tnthornton — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (hasheddan) [reply go.mod:232]: Yep -- good call 👍🏻

Comment (hasheddan) [reply go.mod:232]: As long as k8s bump doesn't impact the types we are using (which it doesn't currently), I think we can just drop support. If we were to use some alpha APIs in the future we would likely need to be more careful 👍🏻 Hopefully we don't have to keep these otel pins around forever, they are currently a result of the intersection of some of our deps.

**Pull request merged**